### Heroes Of Pymoli Data Analysis
* Of the 1163 active players, the vast majority are male (84%). There also exists, a smaller, but notable proportion of female players (14%).

* Our peak age demographic falls between 20-24 (44.8%) with secondary groups falling between 15-19 (18.60%) and 25-29 (13.4%).  
-----

### Note
* Instructions have been included for each segment. You do not have to follow them exactly, but they are included to help you think through the steps.

In [1]:
# Dependencies and Setup
import pandas as pd
import numpy as np
file_to_load = "Resources/purchase_data.csv"
# Read Purchasing File and store into Pandas data frame (two copies)
purchase_data = pd.read_csv(file_to_load)
purchase_analysis = purchase_data.copy()
purchasing_analysis = purchase_data.copy()
top_spenders = purchase_data.copy()
popular_items = purchase_data.copy()
#purchase_data.head(5)
#purchase_analysis

* Display the total number of players


In [2]:
purchase_data.columns

Index(['Purchase ID', 'SN', 'Age', 'Gender', 'Item ID', 'Item Name', 'Price'], dtype='object')

In [3]:
#Total number of Players
a = 'Total Number of Players: ' + str(len(purchase_data['SN'].unique()))
print (a)

Total Number of Players: 576


## Purchasing Analysis (Total)

* Run basic calculations to obtain number of unique items, average price, etc.


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame


In [4]:
#number  of unique items = 183 This works
number_of_unique_items = len(purchase_data['Item ID'].unique())
#average price
average_price = purchase_data ['Price'].mean()
f_average_price = '$' + str(round(average_price,2))
#number of purchases
number_of_purchases = len(purchase_data)
#total revenue
total_revenue = purchase_data ['Price'].sum()
f_total_revenue = '$' + str(round(total_revenue,2))

#print (number_of_unique_items)
#print (f_average_price)
#print(number_of_purchases)
#print (f_total_revenue)


In [5]:
#display summary of purchase data
summary_purch_data_df = pd.DataFrame({"Number of Unique Items":[number_of_unique_items], "Average Price":[f_average_price], 'Number of Purchases':[number_of_purchases],'Total Revenue':[f_total_revenue]})
summary_purch_data_df.head()

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     183         $3.05                  780      $2379.77

## Gender Demographics

* Percentage and Count of Male Players


* Percentage and Count of Female Players


* Percentage and Count of Other / Non-Disclosed




In [6]:
#new table of unique players
num_of_players = len(purchase_data['SN'].unique())
purchase_data.drop_duplicates(subset='SN',keep='first', inplace=True)
gender_summary = purchase_data.groupby(['Gender']).count()
num_of_players
#unique players by gender
gender_summary = gender_summary.rename(columns = {'SN':'Count'})
gender_summary_s = gender_summary[['Count']]
gender_summary_s


Count
Gender                      
Female                    81
Male                     484
Other / Non-Disclosed     11

In [7]:
#players summarywith % by gender
gender_summary_s = gender_summary.loc[:,["Count"]]
gender_summary_s["Persentage of Players"] = (round((gender_summary_s ["Count"])/num_of_players *100, 2)) 
gender_summary_s

Count  Persentage of Players
Gender                                             
Female                    81                  14.06
Male                     484                  84.03
Other / Non-Disclosed     11                   1.91


## Purchasing Analysis (Gender)

* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender




* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [8]:
#purchase_analysis.head()
#purchase_analysis.count()

In [9]:
reduced_purchase_analysis = purchase_analysis.loc[:, ["SN", "Gender", "Purchase ID", "Price"]]
#reduced_purchase_analysis.head(10)
#reduced_purchase_analysis.count()


In [10]:
user_spent = purchase_analysis.groupby(['Gender', 'SN'])['Price'].agg(['sum', 'count'])
#user_spent_df = user_spent.to_frame()
user_spent.head(10)

sum  count
Gender SN                        
Female Adastirin33    4.48      1
       Aerithllora36  8.64      2
       Aethedru70     3.54      1
       Aidain51       3.45      1
       Aiduesu86      4.48      1
       Aillyrin83     2.89      1
       Aisurdru79     3.51      2
       Assilsan72     4.90      1
       Assosia88      1.33      1
       Baelollodeu94  9.03      2

In [11]:
# = reduced_purchase_analysis.groupby(['Gender', 'SN'])['Price'].sum()
user_spent['Average User Purchase'] = user_spent['sum']/user_spent['count']
user_spent.head()

sum  count  Average User Purchase
Gender SN                                               
Female Adastirin33    4.48      1                   4.48
       Aerithllora36  8.64      2                   4.32
       Aethedru70     3.54      1                   3.54
       Aidain51       3.45      1                   3.45
       Aiduesu86      4.48      1                   4.48

In [12]:
#Purchase count by gender, average price of putchase by gender, total purchase $$ by gender, average total purchase per person
spent_by_gender = reduced_purchase_analysis. groupby (['Gender']) ['Price'].agg(['count', 'sum'])
spent_by_gender['Average Purchase Price']=spent_by_gender['sum']/spent_by_gender['count']
spent_by_gender

count      sum  Average Purchase Price
Gender                                                       
Female                   113   361.94                3.203009
Male                     652  1967.64                3.017853
Other / Non-Disclosed     15    50.19                3.346000

In [13]:
#number of users
gender_summary[['Purchase ID']]


Purchase ID
Gender                            
Female                          81
Male                           484
Other / Non-Disclosed           11

In [14]:
#aggregated table
spent_by_gender['Average Total Purchase Per Person']=spent_by_gender['sum']/gender_summary['Purchase ID']
spent_by_gender

count      sum  Average Purchase Price  \
Gender                                                          
Female                   113   361.94                3.203009   
Male                     652  1967.64                3.017853   
Other / Non-Disclosed     15    50.19                3.346000   

                       Average Total Purchase Per Person  
Gender                                                    
Female                                          4.468395  
Male                                            4.065372  
Other / Non-Disclosed                           4.562727

In [15]:
#format result table
spent_by_gender.rename(columns={'count': 'Purchase Count', 'sum': 'Total Purchase Value'}, inplace=True)
spent_by_gender['Average Purchase Price'] = '$' + (spent_by_gender['Average Purchase Price'].astype(float)).round(2).astype(str)
spent_by_gender['Average Total Purchase Per Person'] = '$' + (spent_by_gender['Average Total Purchase Per Person'].astype(float)).round(2).astype(str)
spent_by_gender['Total Purchase Value'] = '$' + (spent_by_gender['Total Purchase Value'].astype(float)).round(2).astype(str)
spent_by_gender.head(20)

Purchase Count Total Purchase Value  \
Gender                                                       
Female                            113              $361.94   
Male                              652             $1967.64   
Other / Non-Disclosed              15               $50.19   

                      Average Purchase Price Average Total Purchase Per Person  
Gender                                                                          
Female                                  $3.2                             $4.47  
Male                                   $3.02                             $4.07  
Other / Non-Disclosed                  $3.35                             $4.56

## Age Demographics

* Establish bins for ages


* Categorize the existing players using the age bins. Hint: use pd.cut()


* Calculate the numbers and percentages by age group


* Create a summary data frame to hold the results


* Optional: round the percentage column to two decimal points


* Display Age Demographics Table


In [16]:
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [17]:
# Create the bins in which Data will be held
bins = [0, 9.9, 14, 19, 24, 29, 34, 39, 200]
# Create the names for the bins
group_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]


In [18]:
#pd.cut
purchase_data ["Age Group Summary"] = pd.cut(purchase_data ["Age"], bins, labels=group_names)
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Group Summary  
0  Extraction, Quickblade Of Trembling Hands   3.53             20-24  
1                          Frenzied Scimitar   1.56               40+  
2                               Final Critic   4.88             20-24  
3                                Blindscythe   3.27             20-24  
4                                       Fury   1.44             20-24

In [19]:
purchase_data_binned = purchase_data.groupby("Age Group Summary").count()
purchase_data_binned

Purchase ID   SN  Age  Gender  Item ID  Item Name  Price
Age Group Summary                                                          
<10                         17   17   17      17       17         17     17
10-14                       22   22   22      22       22         22     22
15-19                      107  107  107     107      107        107    107
20-24                      258  258  258     258      258        258    258
25-29                       77   77   77      77       77         77     77
30-34                       52   52   52      52       52         52     52
35-39                       31   31   31      31       31         31     31
40+                         12   12   12      12       12         12     12

In [20]:
#age bins
purchase_data_binned.reset_index()
players_count = purchase_data_binned['Purchase ID'].sum()
#select only the needed columns, format percentages
purchase_data_binned.drop(purchase_data_binned.columns[[1,2, 3,4,5,6]], axis=1, inplace=True) 
purchase_data_binned['Percentage of Players'] =(((purchase_data_binned['Purchase ID']) /players_count).astype(float)*100).round(2).astype(str) + '%'
print(players_count)
print(purchase_data_binned)

576
                   Purchase ID Percentage of Players
Age Group Summary                                   
<10                         17                 2.95%
10-14                       22                 3.82%
15-19                      107                18.58%
20-24                      258                44.79%
25-29                       77                13.37%
30-34                       52                 9.03%
35-39                       31                 5.38%
40+                         12                 2.08%


In [21]:
purchase_data_binned=purchase_data_binned.rename(columns={"Purchase ID":"Players Count"})
purchase_data_binned

Players Count Percentage of Players
Age Group Summary                                     
<10                           17                 2.95%
10-14                         22                 3.82%
15-19                        107                18.58%
20-24                        258                44.79%
25-29                         77                13.37%
30-34                         52                 9.03%
35-39                         31                 5.38%
40+                           12                 2.08%

## Purchasing Analysis (Age)

* Bin the purchase_data data frame by age


* Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below


* Create a summary data frame to hold the results


* Optional: give the displayed data cleaner formatting


* Display the summary data frame

In [22]:
# using df purchasing_analysis
# Create the bins in which Data will be held
bins = [0, 9.9, 14, 19, 24, 29, 34, 39, 200]
# Create the names for the bins
group_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]
#pd.cut
purchasing_analysis ["Age Group Summary"] = pd.cut(purchasing_analysis ["Age"], bins, labels=group_names)
purchasing_analysis.head(20)

Purchase ID             SN  Age                 Gender  Item ID  \
0             0        Lisim78   20                   Male      108   
1             1    Lisovynya38   40                   Male      143   
2             2     Ithergue48   24                   Male       92   
3             3  Chamassasya86   24                   Male      100   
4             4      Iskosia90   23                   Male      131   
5             5        Yalae81   22                   Male       81   
6             6      Itheria73   36                   Male      169   
7             7    Iskjaskst81   20                   Male      162   
8             8      Undjask33   22                   Male       21   
9             9    Chanosian48   35  Other / Non-Disclosed      136   
10           10      Inguron55   23                   Male       95   
11           11   Haisrisuir60   23                   Male      162   
12           12   Saelaephos52   21                   Male      116   
13           13    Assjaskan73   22                   Male        4   
14           14    Saesrideu94   35                   Male      165   
15           15      Lisassa64   21                 Female       98   
16           16      Lisirra25   20                   Male       40   
17           17      Zontibe81   21                   Male      161   
18           18      Reunasu60   22                 Female       82   
19           19      Chamalo71   30                   Male       89   

                                    Item Name  Price Age Group Summary  
0   Extraction, Quickblade Of Trembling Hands   3.53             20-24  
1                           Frenzied Scimitar   1.56               40+  
2                                Final Critic   4.88             20-24  
3                                 Blindscythe   3.27             20-24  
4                                        Fury   1.44             20-24  
5                                   Dreamkiss   3.61             20-24  
6      Interrogator, Blood Blade of the Queen   2.18             35-39  
7                               Abyssal Shard   2.67             20-24  
8                                   Souleater   1.10             20-24  
9                Ghastly Adamantite Protector   3.58             35-39  
10                      Singed Onyx Warscythe   4.74             20-24  
11                              Abyssal Shard   2.67             20-24  
12                    Renewed Skeletal Katana   4.18             20-24  
13                         Bloodlord's Fetish   1.70             20-24  
14                Bone Crushing Silver Skewer   4.86             35-39  
15                Deadline, Voice Of Subtlety   2.89             20-24  
16                              Second Chance   2.52             20-24  
17                                     Devine   1.76             20-24  
18                                    Nirvana   4.90             20-24  
19          Blazefury, Protector of Delusions   4.64             30-34

In [23]:
purch_stats_binned = purchasing_analysis.groupby('Age Group Summary', as_index=False).agg({'Purchase ID': 'count', 'Price': sum})
#type(purch_stats_binned)
purch_stats_binned


Age Group Summary  Purchase ID    Price
0               <10           23    77.13
1             10-14           28    82.78
2             15-19          136   412.89
3             20-24          365  1114.06
4             25-29          101   293.00
5             30-34           73   214.00
6             35-39           41   147.67
7               40+           13    38.24

In [24]:
person_stats_binned = purchasing_analysis.groupby(['Age Group Summary', 'SN'], as_index=False).agg({'Purchase ID': 'count', 'Price': sum})
person_stats_binned.head(20)

Age Group Summary             SN  Purchase ID  Price
0                <10   Anallorgue57            2   5.70
1                <10     Chadjask77            1   4.93
2                <10   Chanossast57            1   4.32
3                <10   Eurithphos97            2   4.55
4                <10        Eusri44            1   3.09
5                <10  Frichossala54            1   3.08
6                <10   Haillyrgue51            3   9.50
7                <10       Heudai45            1   3.47
8                <10      Ililsan66            1   2.85
9                <10        Ilmol66            2   6.17
10               <10    Quaecjask96            1   4.40
11               <10     Quilassa66            1   4.23
12               <10      Seuthep89            1   2.05
13               <10        Sisur91            1   4.19
14               <10     Sondadar26            2   6.47
15               <10     Sondossa69            1   4.48
16               <10     Tyeurith29            1   3.65
17             10-14     Aeralria27            1   4.09
18             10-14     Aillyrin83            1   2.89
19             10-14  Chanirrasta87            1   4.94

In [25]:
person_stats1_binned = person_stats_binned.groupby('Age Group Summary', as_index=False).agg({'Purchase ID': sum, 'Price': sum, 'SN': 'count'})
person_stats1_binned

Age Group Summary  Purchase ID    Price   SN
0               <10           23    77.13   17
1             10-14           28    82.78   22
2             15-19          136   412.89  107
3             20-24          365  1114.06  258
4             25-29          101   293.00   77
5             30-34           73   214.00   52
6             35-39           41   147.67   31
7               40+           13    38.24   12

In [26]:
#average spent per person in each age group
person_stats1_binned['Average Purchase Per Person'] = person_stats1_binned['Price']/person_stats1_binned['SN']
person_stats1_binned['Average Purchase Price']=person_stats1_binned['Price']/person_stats1_binned['Purchase ID']
#rename columns
person_stats1_binned.rename(columns={'Purchase ID': 'Purchase Count', 'Price': 'Total Purchase Value', 'SN':'Players Count'}, inplace=True)
person_stats1_binned

Age Group Summary  Purchase Count  Total Purchase Value  Players Count  \
0               <10              23                 77.13             17   
1             10-14              28                 82.78             22   
2             15-19             136                412.89            107   
3             20-24             365               1114.06            258   
4             25-29             101                293.00             77   
5             30-34              73                214.00             52   
6             35-39              41                147.67             31   
7               40+              13                 38.24             12   

   Average Purchase Per Person  Average Purchase Price  
0                     4.537059                3.353478  
1                     3.762727                2.956429  
2                     3.858785                3.035956  
3                     4.318062                3.052219  
4                     3.805195                2.900990  
5                     4.115385                2.931507  
6                     4.763548                3.601707  
7                     3.186667                2.941538

In [27]:
#format and drop columns
person_stats1_binned.drop(person_stats1_binned.columns[[3]], axis=1, inplace=True) 


In [28]:
person_stats1_binned['Total Purchase Value'] = '$'+(person_stats1_binned['Total Purchase Value']).round(2).astype(str)
person_stats1_binned['Average Purchase Per Person'] = '$'+(person_stats1_binned['Average Purchase Per Person']).round(2).astype(str)
person_stats1_binned['Average Purchase Price'] = '$'+(person_stats1_binned['Average Purchase Price']).round(2).astype(str)
person_stats1_binned

Age Group Summary  Purchase Count Total Purchase Value  \
0               <10              23               $77.13   
1             10-14              28               $82.78   
2             15-19             136              $412.89   
3             20-24             365             $1114.06   
4             25-29             101               $293.0   
5             30-34              73               $214.0   
6             35-39              41              $147.67   
7               40+              13               $38.24   

  Average Purchase Per Person Average Purchase Price  
0                       $4.54                  $3.35  
1                       $3.76                  $2.96  
2                       $3.86                  $3.04  
3                       $4.32                  $3.05  
4                       $3.81                   $2.9  
5                       $4.12                  $2.93  
6                       $4.76                   $3.6  
7                       $3.19                  $2.94

In [29]:
#transpose columns 0,1,3,4,2
person_stats1_binned.iloc[:,[0,1,4,2,3]]

Age Group Summary  Purchase Count Average Purchase Price  \
0               <10              23                  $3.35   
1             10-14              28                  $2.96   
2             15-19             136                  $3.04   
3             20-24             365                  $3.05   
4             25-29             101                   $2.9   
5             30-34              73                  $2.93   
6             35-39              41                   $3.6   
7               40+              13                  $2.94   

  Total Purchase Value Average Purchase Per Person  
0               $77.13                       $4.54  
1               $82.78                       $3.76  
2              $412.89                       $3.86  
3             $1114.06                       $4.32  
4               $293.0                       $3.81  
5               $214.0                       $4.12  
6              $147.67                       $4.76  
7               $38.24                       $3.19

## Top Spenders

* Run basic calculations to obtain the results in the table below


* Create a summary data frame to hold the results


* Sort the total purchase value column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [30]:
top_spenders.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [31]:
#calculate
top_spenders_grouped = top_spenders.groupby(['SN']).agg({'Item ID':'count','Price': sum})
top_spenders_grouped['Average Purchase Price'] =  top_spenders_grouped['Price']/top_spenders_grouped['Item ID']                                                    
top_spenders_grouped.head()


Item ID  Price  Average Purchase Price
SN                                                  
Adairialis76        1   2.28                    2.28
Adastirin33         1   4.48                    4.48
Aeda94              1   4.91                    4.91
Aela59              1   4.32                    4.32
Aelaria33           1   1.79                    1.79

In [32]:
#rename columns
top_spenders_grouped.rename(columns={'Item ID': 'Purchase Count', 'Price': 'Total Purchase Value'}, inplace=True)
top_spenders_grouped.head()

Purchase Count  Total Purchase Value  Average Purchase Price
SN                                                                        
Adairialis76               1                  2.28                    2.28
Adastirin33                1                  4.48                    4.48
Aeda94                     1                  4.91                    4.91
Aela59                     1                  4.32                    4.32
Aelaria33                  1                  1.79                    1.79

In [33]:
#sort
top_spenders_summary = top_spenders_grouped.sort_values(by='Total Purchase Value', ascending = False)
top_spenders_summary.head()

Purchase Count  Total Purchase Value  Average Purchase Price
SN                                                                       
Lisosia93                 5                 18.96                3.792000
Idastidru52               4                 15.45                3.862500
Chamjask73                3                 13.83                4.610000
Iral74                    4                 13.62                3.405000
Iskadarya95               3                 13.10                4.366667

In [34]:
#format amounts
top_spenders_summary['Total Purchase Value']='$'+(top_spenders_summary['Total Purchase Value'].astype(float)).round(2).astype(str)
top_spenders_summary['Average Purchase Price'] = '$' + (top_spenders_summary['Average Purchase Price'].astype(float)).round(2).astype(str)
top_spenders_summary.head()

Purchase Count Total Purchase Value Average Purchase Price
SN                                                                     
Lisosia93                 5               $18.96                  $3.79
Idastidru52               4               $15.45                  $3.86
Chamjask73                3               $13.83                  $4.61
Iral74                    4               $13.62                   $3.4
Iskadarya95               3                $13.1                  $4.37

## Most Popular Items

* Retrieve the Item ID, Item Name, and Item Price columns


* Group by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value


* Create a summary data frame to hold the results


* Sort the purchase count column in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the summary data frame



In [35]:
#popular items
popular_items.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [36]:
#Group by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value
popular_items_groups=popular_items.groupby(['Item ID', 'Item Name'], as_index=False).agg({'SN':'count', 'Price': 'sum'})
popular_items_groups.head()

Item ID           Item Name  SN  Price
0        0            Splinter   4   5.12
1        1            Crucifer   3   9.78
2        2             Verdict   6  14.88
3        3        Phantomlight   6  14.94
4        4  Bloodlord's Fetish   5   8.50

In [37]:
#rename columns 
popular_items_groups.rename(columns={'SN':'Purchase Count','Price': 'Total Purchase Value'}, inplace=True)
popular_items_groups.head()

Item ID           Item Name  Purchase Count  Total Purchase Value
0        0            Splinter               4                  5.12
1        1            Crucifer               3                  9.78
2        2             Verdict               6                 14.88
3        3        Phantomlight               6                 14.94
4        4  Bloodlord's Fetish               5                  8.50

In [38]:
popular_items_groups['Item Price']=popular_items_groups['Total Purchase Value']/popular_items_groups['Purchase Count']
popular_items_groups.head()

Item ID           Item Name  Purchase Count  Total Purchase Value  \
0        0            Splinter               4                  5.12   
1        1            Crucifer               3                  9.78   
2        2             Verdict               6                 14.88   
3        3        Phantomlight               6                 14.94   
4        4  Bloodlord's Fetish               5                  8.50   

   Item Price  
0        1.28  
1        3.26  
2        2.48  
3        2.49  
4        1.70

In [43]:
#Sort the purchase count column in descending order and format
popular_items_groups2 = popular_items_groups.sort_values(by='Purchase Count',ascending = False)
popular_items_groups2['Total Purchase Value']='$'+(popular_items_groups2['Total Purchase Value'].astype(float)).round(2).astype(str)
popular_items_groups2['Item Price']='$'+(popular_items_groups2['Item Price'].astype(float)).round(2).astype(str)
popular_items_groups2.head()

Item ID                                     Item Name  Purchase Count  \
177      178  Oathbreaker, Last Hope of the Breaking Storm              12   
144      145                          Fiery Glass Crusader               9   
107      108     Extraction, Quickblade Of Trembling Hands               9   
81        82                                       Nirvana               9   
19        19                 Pursuit, Cudgel of Necromancy               8   

    Total Purchase Value Item Price  
177               $50.76      $4.23  
144               $41.22      $4.58  
107               $31.77      $3.53  
81                 $44.1       $4.9  
19                 $8.16      $1.02

## Most Profitable Items

* Sort the above table by total purchase value in descending order


* Optional: give the displayed data cleaner formatting


* Display a preview of the data frame



In [44]:
## Sort the purchase count column in descending order
popular_items_groups1 = popular_items_groups.sort_values(by='Total Purchase Value',ascending = False)
popular_items_groups1['Total Purchase Value']='$'+(popular_items_groups1['Total Purchase Value'].astype(float)).round(2).astype(str)
popular_items_groups1['Item Price']='$'+(popular_items_groups1['Item Price'].astype(float)).round(2).astype(str)
popular_items_groups1.head()

Item ID                                     Item Name  Purchase Count  \
177      178  Oathbreaker, Last Hope of the Breaking Storm              12   
81        82                                       Nirvana               9   
144      145                          Fiery Glass Crusader               9   
91        92                                  Final Critic               8   
102      103                                Singed Scalpel               8   

    Total Purchase Value Item Price  
177               $50.76      $4.23  
81                 $44.1       $4.9  
144               $41.22      $4.58  
91                $39.04      $4.88  
102                $34.8      $4.35

# My observations:
# 1. Females represent only 14% of the players demographycs
# 2. Females spend about 7% more money then males
# 3. Many of the most popular items happen to generate the most profit as well
# 4. About player's age: Player's gender is not as important as the player's age. #In fact, age is the most important factor defining the player's purchasing behaviour. 
# Of the 576 players total, 258 are in the 20-24 year old group. That is 44.7%
# 442 players belong to the 15-29 year old group. That is 76.7%